dev/com.ibm.ws.microprofile.faulttolerance.2.0/src/com/ibm/ws/microprofile/faulttolerance20/state/impl/TimeoutStateImpl.java

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.faulttolerance20.state.impl;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import com.ibm.ws.microprofile.faulttolerance.spi.TimeoutPolicy;
+import com.ibm.ws.microprofile.faulttolerance20.state.TimeoutState;
+
+public class TimeoutStateImpl implements TimeoutState {
+
+    private final ScheduledExecutorService executorService;
+    private final TimeoutPolicy policy;
+    private TimeoutResult result = TimeoutResult.NEW;
+    private Runnable timeoutCallback;
+
+    public TimeoutStateImpl(ScheduledExecutorService executorService, TimeoutPolicy policy) {
+        super();
+        this.executorService = executorService;
+        this.policy = policy;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public synchronized void start(Runnable timeoutCallback) {
```

Comment:
switch to synchronized blocks such that the entry-exit trace is outside

Reply:
done